stylix/mk-target.nix

```diff
@@ -221,7 +228,16 @@ let
 in
 {
   imports = [
-    { options.stylix.targets.${name} = extraOptions; }
+    (
+      { config, ... }:
+      {
+        options.stylix.targets.${name} =
+          if builtins.isFunction extraOptions then
+            extraOptions config.lib.stylix
```

Comment:
I don't think this will scale well. In the future, you may want other things to be provided to the `extraOptions` function too, e.g. `cfg` or `opts` may be useful here.

I would recommend passing it as an attr, e.g.
`{ stylixLib = config.lib.stylix; }`.

Long term, this would be best using `functionArgs`, similar to `configElements` below. But initially, it'd be fine to just pass the attr unconditionally.